adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/VertxBasedMqttProtocolAdapter.java

```diff
@@ -162,8 +169,12 @@ private void handleEndpointConnection(final MqttEndpoint endpoint) {
             endpoint.reject(MqttConnectReturnCode.CONNECTION_REFUSED_SERVER_UNAVAILABLE);
 
         } else {
-
-            endpoint.publishHandler(message -> {
+            final AtomicBoolean clientConnected = new AtomicBoolean(false);
```

Comment:
I think that would be best, in particular because we would not see it being exposed on the interface before 3.4.0. FINAL, I guess ...